### ISSUE TITLE
No diagnostic reported for non-primitive casts

### ISSUE THREAD

**Issue by lnicola** [rust-lang/rust-analyzer#17897]
```rust
_ = ((), ()) as ();
```

**Comment by ShoyuVanilla** [comment]
It seems that we aren't making any diagnostics here yet

https://github.com/rust-lang/rust-analyzer/blob/07659783fdfd4ec0a0bffa93017e33e31e567e42/crates/hir-ty/src/infer/cast.rs#L16-L34
